hydrolib/core/io/ini/serializer.py

```diff
@@ -0,0 +1,246 @@
+from dataclasses import dataclass
+from functools import reduce
+from hydrolib.core.io.ini.models import CommentBlock, Document, Property, Section
+from hydrolib.core.basemodel import BaseModel
+from itertools import chain, count, repeat
+from pathlib import Path
+from typing import Any, Iterable, List, Optional, Sequence, Tuple, Union
+
+
+class SerializerConfig(BaseModel):
+    """SerializerConfig defines the configuration options of the Serializer
+
+    Attributes:
+        section_indent (int):
+            The number of spaces with which whole sections should be indented.
+            Defaults to 0.
+        property_indent (int):
+            The number of spaces with which properties should be indented relative to
+            the section header (i.e. the full indent equals the section_indent plus
+            property_indent). Defaults to 4.
+        datablock_indent (int):
+            The number of spaces with which datablock rows are indented relative to
+            the section header (i.e. the full indent equals the section_indent plus
+            datablock_indent). Defaults to 8.
+        datablock_spacing (int):
+            The number of spaces between datablock columns. Note that there might be
+            additional offset to ensure . is lined out. Defaults to 4.
+        comment_delimiter (str):
+            The character used to delimit comments.
+    """
+
+    section_indent: int = 0
+    property_indent: int = 4
+    datablock_indent: int = 8
+    datablock_spacing: int = 4
+    comment_delimiter: str = "#"
+
+    @property
+    def total_property_indent(self):
+        return self.section_indent + self.property_indent
+
+    @property
+    def total_datablock_indent(self):
+        return self.section_indent + self.datablock_indent
+
+
+@dataclass
+class _Lengths:
```

Comment:
So each section has its own value & comment offset? 
Is it desirable that each section has a different offsets?

Reply:
Yup, that's the idea. That was how I understand it, however it would be easy to change in the future if we do need to to change it.

Reply:
Ok, I always thought the offsets apply to the whole file.